lib/tapioca/gem/pipeline.rb

```diff
@@ -126,35 +126,38 @@ def symbol_in_payload?(symbol_name)
         @payload_symbols.include?(symbol_name)
       end
 
-      # this looks something like:
-      # "(eval at /path/to/file.rb:123)"
-      # and we are just interested in the "/path/to/file.rb" part
-      EVAL_SOURCE_FILE_PATTERN = T.let(/\(eval at (.+):\d+\)/, Regexp)
-
       sig { params(name: T.any(String, Symbol)).returns(T::Boolean) }
       def constant_in_gem?(name)
-        return true unless Object.respond_to?(:const_source_location)
-
-        source_file, _ = Object.const_source_location(name)
-        return true unless source_file
-        # If the source location of the constant is "(eval)", all bets are off.
-        return true if source_file == "(eval)"
+        source_file, _ = const_source_location(name)
 
-        # Ruby 3.3 adds automatic definition of source location for evals if
-        # `file` and `line` arguments are not provided. This results in the source
-        # file being something like `(eval at /path/to/file.rb:123)`. We try to parse
-        # this string to get the actual source file.
-        source_file = source_file.sub(EVAL_SOURCE_FILE_PATTERN, "\\1")
+        # If the source location of the constant isn't available or is "(eval)", all bets are off.
+        return true if source_file.nil? || source_file == "(eval)"
 
         gem.contains_path?(source_file)
       end
 
-      sig { params(method: UnboundMethod).returns(T::Boolean) }
-      def method_in_gem?(method)
-        source_location = method.source_location&.first
-        return false if source_location.nil?
+      sig { params(method_name: Symbol, owner: Module).returns(T.any([String, Integer], NilClass, T::Boolean)) }
+      def method_definition_in_gem(method_name, owner)
```

Comment:
The name is less-than-obvious, but I can't think of a better one. How about we add a doc comment? The return value is particularly tricky to grok

```suggestion
      # @returns - The source file and line number, if this method was defined in this pipeline's gem
                 - `nil` if the source location is unknown
                 - `true` if the method was defined via `eval` without a known file/line number
                 - `false` if this method wasn't defined by this gem
      sig { params(method_name: Symbol, owner: Module).returns(T.any([String, Integer], NilClass, T::Boolean)) }
      def method_definition_in_gem(method_name, owner)
```